libcxx/include/__exception/exception_ptr.h

```diff
@@ -51,11 +71,25 @@ class _LIBCPP_EXPORTED_FROM_ABI exception_ptr {
 template <class _Ep>
 _LIBCPP_HIDE_FROM_ABI exception_ptr make_exception_ptr(_Ep __e) _NOEXCEPT {
 #  ifndef _LIBCPP_HAS_NO_EXCEPTIONS
+#    if _LIBCPP_AVAILABILITY_HAS_INIT_PRIMARY_EXCEPTION
+  using _Ep2 = __decay_t<_Ep>;
+  void* __ex = exception_ptr::__init_native_exception(
+      sizeof(_Ep), const_cast<std::type_info*>(&typeid(_Ep)), exception_ptr::__dest_thunk<_Ep2>);
```

Comment:
```suggestion
  void* __ex = exception_ptr::__init_native_exception(
      sizeof(_Ep), const_cast<std::type_info*>(&typeid(_Ep)), [](void* __p) {
          std::__destroy_at(static_cast<_Ep>(__p));
      });
```

Then you don't need `__dest_thunk` at all.

Reply:
Done

Reply:
Using lambda here requires C++11, so i guarded all this with `__cplusplus >= 201103L`